server/genserver.go

```diff
@@ -222,11 +222,25 @@ func NewWithMiddleware(c Controller, addr string, m []func(http.Handler) http.Ha
 		r = r.WithContext(ctx)
 	})
 	{{end}}
+	return router
+}
 
-	handler := withMiddleware("{{.Title}}", router, m)
-	return &Server{Handler: handler, addr: addr, l: l}
+// NewWithMiddleware returns a Server that implemenets the Controller interface. It runs the
+// middleware after the built-in middleware (e.g. logging), but before the controller methods.
+// The middleware is executed in the order specified. The server will start when "Serve" is called.
+func NewWithMiddleware(c Controller, addr string, m []func(http.Handler) http.Handler) *Server {
+	router := newRouter(c)
+
+	return AttachMiddleware(router, addr, m)
 }
-`
+
+// AttachMiddleware attaches the given middleware to the router 
+func AttachMiddleware(router http.Handler, addr string, m []func(http.Handler) http.Handler) *Server {
```

Comment:
here I'd also change this from `http.Handler` to `*mux.Router` and explain in the comment that it is meant to be used in conjuction with `NewServer`. I might also add in the comment that it attaches custom middleware passed as arguments as well as built-in wag middleware for logging, handling panics, and tracing.